packages/storybook/src/generators/configuration/util-functions.ts

```diff
@@ -435,3 +496,35 @@ export function addBuildStorybookToCacheableOperations(tree: Tree) {
     },
   }));
 }
+
+export function projectIsRootProjectInNestedWorkspace(root: string) {
+  return relative(workspaceRoot, root).length === 0;
+}
+
+export function workspaceInNestedMode(tree: Tree) {
+  return tree.exists('src') && tree.exists('project.json');
```

Comment:
I think we just need to check for root `project.json`, since `src` is not guaranteed. For example, Next.js apps do not have `src`.

Instead of `workspaceInNestedMode` I'd call it `workspaceHasRootProject` since nesting may or may not be true (i.e. there will only be nesting if the user generates a lib with `project.json` in its directory, which doesn't happen out of the box).